assets/components/Buttons.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+export type IPropsVariant = 'primary' | 'secondary' | 'tertiary';
+export type IPropsSize = 'small' | 'medium' | 'large';
+
+export interface IPropsButtonBase {
+    value: string;
+    variant?: IPropsVariant;
+    size?: IPropsSize;
+    id?: string;
+    className?: string;
+    disabled?: boolean;
+    'data-test-id'?: string;
+}
+
+interface IPropsButton extends IPropsButtonBase {
+    type?: 'button',
+    onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
+}
+
+interface IPropsSubmit extends IPropsButtonBase {
+    type?: 'submit',
+    onClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
+}
+
+interface IPropsReset extends IPropsButtonBase {
+    type?: 'reset',
```

Comment:
I guess here the type must be `reset` so should be probably just `type: 'reset'`,
I'm also wondering if we have a usecase where we wouldn't use it with `onClick`, might be better to make it mandatory if not to get this checked.
same applies to submit